tools/script_extractor.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python3
+
+import argparse
+
+#TODO:
+# - Add arguments to all functions (have fun lol)
+# - implement text
+# - (Possibly) add new type of word that looks for matches in the sym file
+# - enter to continually generate more lines of the script
+# - full script mode?
+# - Make pretty!!
+
+def decodeLine(scriptList, game_data, loc, ignore_broken):
+	currLine = scriptList[game_data[loc]]
+	ret = "\trun_script " + currLine[0] + "\n"
+	loc+=1
+	quit = currLine[2]
+	for c in currLine[1]:
+		if c == "b":
+			ret += "\tdb $" + format(game_data[loc],"02x") + "\n"
+			loc += 1
+		elif c == "w":
+			ret += "\tdb $" + format((game_data[loc] + (game_data[loc+1]<<8)),"04x") + "\n"
+			loc += 2
+		elif c == "t":
+			ret += "\ttx Text" + format((game_data[loc] + (game_data[loc+1]<<8)),"04x") + "\n"
+			loc += 2
+		elif c == "q":
+			print("haven't updated data for this yet")
+			if not ignore_broken:
+				quit = True
+	return (loc, ret, quit)
+			
+
+def main():
+	scriptList = createList()
+
+	with open("tcg.gbc", "rb") as file:
+	    game_data = file.read()
+	loc = 0xd52e
+	auto = True
+	end = False
+	ignore_broken = True
+	script = ""
+	if game_data[loc] != 0xe7:
+		print("Error: first byte was not start_script")
+	else:
+ 		
+		# TODO this is hacky please don't do this 
+		ls = format(loc,"04x")
+		lsa = format(loc-0x8000,"04x")
+		print("OWSequence_" + ls + ": ; " + ls + " (3:" + lsa + ")" )
+		loc += 1
+		print("\tstart_script")
+	while not end:
+		loc, outstr, end = decodeLine(scriptList,game_data,loc,ignore_broken)
+		outstr = outstr[:-1] # [:-1] strips the newline at the end
+		if auto:
+			print(outstr)
+		else:
+			input(outstr)
+	print("; " + hex(loc))
+		
+
+def createList(): # this is a func just so all this can go at the bottom
```

Comment:
Nice database you made!
And, I think that maybe we can make use of your argument-information, and we could generate automatically script macros for code simplicity.
For example,

script_PrintTextString: MACRO
	run_script OWScript_PrintTextString
	tx \1
ENDM

and using like this:

script_PrintTextString Text05e5

I have not looked into its script part before,
and now I think we can improve it with macros.

Reply:
I do like that, however I'm worried that with how many scripts there are, we would have tons of macros. Another thing we could do is a general macro that lets you do

`script OWScript_PrintTextString, Text05e5`
 
However, there are some OWScripts that are 5+ args long [And I'm not entirely sure if you can/how to allow args to be 1 byte or 2).

I think eventually extra scripting macros is the way to go - however should we start off with that many or find a middle ground on which ones to give special macros to?

Reply:
Yeah. Those scripts use variant args, so you can state each macro for each script.
I mean, for each script, one specific macro.

Like,
script_StartBattle 2, WATERFRONT_POKEMON_DECK_ID, MUSIC_DUEL_THEME_1
(Note that there is no space between 'script' and script name)

With that database, it's easy to generate them all.

And actually, I'd like to use pre-processing way, which would be your extra script macros?

Anyway, it was just my opinion.
I don't care which approach to use if it works!